File: 后台/trade/types/address.go
package types

type AddrAddReq struct {
	Name      string `json:"name"`
	Tel       string `json:"tel"`
	Province  string `json:"province"`
	City      string `json:"city"`
	District  string `json:"area"`
	Address   string `json:"detailArea"`
	IsDefault int    `json:"isDefault"`
}

type AddrInfoResp struct {
	ID        int    `json:"id"`
	UserID    int    `json:"userID"`
	Name      string `json:"name"`
	Tel       string `json:"tel"`
	Province  string `json:"province"`
	City      string `json:"city"`
	District  string `json:"area"`
	Address   string `json:"detailArea"`
	IsDefault int    `json:"isDefault"`
}

type AddrUpdateReq struct {
	ID        int    `url:"id" json:"id"`
	Name      string `json:"name"`
	Tel       string `json:"tel"`
	Province  string `json:"province"`
	City      string `json:"city"`
	District  string `json:"area"`
	Address   string `json:"detailArea"`
	IsDefault int    `json:"isDefault"`
}

type AddrAddResp struct {
	ID int `json:"id"`
}

type UpdateDefaultReq struct {
	OldDefault int `json:"oldDefault"`
	NewDefault int `json:"newDefault"`
}

File: 前台/uni-trade-main/src/api/address.js
import httpInstance from '@/utils/https'
import { useUserStore } from '@/store/userStore'

//获取地址列表
export const getAddressListAPI = () => {
  const userStore = useUserStore()
  return httpInstance({
    url: '/address',
    method: 'get',
    headers: {
      Authorization: `${userStore.userInfo.token}`
    }
  })
}

// 添加新地址
export const addAddressAPI = (data) => {
  const userStore = useUserStore()
  return httpInstance({
    url: '/address',
    method: 'post',
    headers: {
      Authorization: `${userStore.userInfo.token}`
    },
    data: {
      name: data.name,
      province: data.province,
      city: data.city,
      area: data.area,
      detailArea: data.detailArea,
      tel: data.tel,
      isDefault: data.isDefault
    }
  })
}

// 修改地址
export const updateAddressAPI = (id, updatedData) => {
  const userStore = useUserStore()
  return httpInstance({
    url: `/address/${id}`,
    method: 'put',
    data: updatedData,
    headers: {
      Authorization: `${userStore.userInfo.token}`
    }
  })
}

// 修改默认地址
export const setDefaultAddressAPI = (data) => {
  const userStore = useUserStore()
  return httpInstance({
    url: `/address/setDefault/${data.newAddressId}`,
    method: 'put',
    data: {
      oldDefault: data.oldAddressId,
      newDefault: data.newAddressId
    },
    headers: {
      Authorization: `${userStore.userInfo.token}`
    }
  })
}

// 删除地址
export const deleteAddressAPI = (id) => {
  const userStore = useUserStore()
  return httpInstance({
    url: `/address/${id}`,
    method: 'delete',
    headers: {
      Authorization: `${userStore.userInfo.token}`
    }
  })
}
